### PR TITLE
Notification and ShiftView

### PR DESCRIPTION
**Pull request by Khazanah-spec**
Have mostly completed NotificationGUI and ShiftViewHRGUI frontend and some of the backend as well as functions that populate lists. Creating additional classes for NotificationGUI for different users is still a work in progress.

### CODE REVIEW

**Review by williamb700 — DISMISSED** [review]
Consider using the factory design pattern to create a notification builder/modifier. Also consider implementing an MVP for the GUI.

File: src/main/java/NotificationHRGUI.java [hunk 1]
```diff
@@ -24,43 +25,65 @@ public class NotificationGUI extends JFrame implements ActionListener {
     public JList<String> resolvedNotificationList;
 
 
-    public NotificationGUI() {
+    public NotificationHRGUI(User user) {
```
Comment (mding010705) [inline src/main/java/NotificationHRGUI.java:28]: replace User user with int userId (outer layer cannot skip intermediate layers and directly depend on entity)

File: src/main/java/NotificationHRGUI.java [hunk 2]
```diff
@@ -24,43 +25,65 @@ public class NotificationGUI extends JFrame implements ActionListener {
...
+                    resolvedNotifications.add(n);
+                }
+            }
+            else{
+                if (n.getSenderId() == user.getEmployeeNum() || n.getRecipientId() == user.getEmployeeNum()){
+                    unresolvedNotifications.add(n);
+                }
+            }
+        }
+        Notification[][] noti = new Notification[][] {Notification.sortByCreatedDate(unresolvedNotifications), Notification.sortByCreatedDate(resolvedNotifications)};
```
Comment (mding010705) [inline src/main/java/NotificationHRGUI.java:62]: use case (not gui) class has to handle operations dealing directly with entities

File: src/main/java/ShiftViewHRGUI.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.util.Objects;
+
+public class ShiftViewHRGUI extends JFrame implements ActionListener {
```
[the remark this answers is not in the document]
Comment (mding010705) [reply src/main/java/ShiftViewHRGUI.java:9]: Should implement Page interface

File: src/main/java/NotificationHRGUI.java (hunk 1 above)
Comment (mding010705) [reply src/main/java/NotificationHRGUI.java:28]: should implement Page interface

**Review by mding010705 — DISMISSED** [review]
ShiftViewHR obviously needs some work, but you're on the right track. Please make your gui pages implement Page and fix the CA violations.

File: src/Notification.java [hunk 4]
```diff
@@ -1,4 +1,7 @@
+import net.bytebuddy.build.HashCodeAndEqualsPlugin;
+
 import java.time.LocalDateTime;
+import java.util.ArrayList;
 import java.util.Interactor;
 import java.util.NotificationDatabaseInteractor;
```
Comment (mding010705) [inline src/Notification.java:7]: Notification should implement serializable

**Pull request merged**